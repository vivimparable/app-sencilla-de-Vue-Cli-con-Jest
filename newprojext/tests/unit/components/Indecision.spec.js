import { shallowMount } from "@vue/test-utils";
import Indecision from '@/components/IndecisionComponent.vue'
import { text } from "@fortawesome/fontawesome-svg-core";

describe('Componente Indecision.vue',()=>{

    let wrapper;
    let clgSpy;

beforeEach(()=>{
    wrapper= shallowMount(Indecision);
    clgSpy = jest.spyOn(console,'log');
    //jest.clearAllMocks()


    global.fetch = jest.fn(()=> Promise.resolve({
        json:()=>Promise.resolve({
                answer:'yes',
                forced:false,
                image:'https://yesno.wtf/assets/yes/2.gif'
            })
        
    }));



})

    test('Snapshot',()=>{
        expect(wrapper.html()).toMatchSnapshot();
    })


    test('Verificar H1',()=>{

        const wrapper = shallowMount(Indecision,{
            props:{
                title:"Â¡Welcome to the indecision game!"
            }
        })

        const dato1 = wrapper.find('h1').text();
       
            expect(dato1).toBe(wrapper.props().title)
    })  

    test('Input a la hora de dispar no suceda nada',async()=>{
        
        const getAnswerSpy = jest.spyOn( wrapper.vm,'peticion' )

        const input1 = wrapper.find('input');
        
        await input1.setValue('Hola Mundo');

            expect(clgSpy).toHaveBeenCalled();
            expect(getAnswerSpy).toHaveBeenCalledTimes(0);

        
    })

    test('Debe llamar el getAnswer',async ()=>{

        const getAnswer = jest.spyOn(wrapper.vm, 'peticion');

        const inputes = wrapper.find('input');

        await inputes.setValue('?')
            expect(getAnswer).toBeCalledTimes(1)
        
    })

    test('Pruebas en getAnswer',async()=>{


        await  wrapper.vm.peticion();
        const img = wrapper.find('img');
            expect(img).toBeTruthy();
            expect(wrapper.vm.answer).toBe('yes'); 
        
    })

    test('Prueba getanswer ---Fallo API',async()=>{

        fetch.mockImplementationOnce(()=>Promise.reject('DOWN API'))


        await wrapper.vm.peticion();
        
            expect(wrapper.vm.answer).toBe('Error Api')
            expect(wrapper.vm.img).toBeFalsy();

        //Mensaje de error
        const errorMsg= wrapper.find('h2').text();
            expect(errorMsg).toBe('ERROR :(')

    })


})