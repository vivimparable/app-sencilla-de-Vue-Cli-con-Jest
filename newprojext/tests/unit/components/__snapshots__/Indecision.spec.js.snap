// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente Indecision.vue Snapshot 1`] = `
<div>
  <h1></h1>
  <!--v-if-->
  <p>¿Tienes dudas acerca de tu vida? Pues qué mejor manera que saberlo a través de memes. Adelante, haz tu pregunta y a ver que te depara el futuro.</p>
  <div class="bg-dark"></div>
  <div class="indecision-container"><input placeholder="¡Preguntame algo!" type="text" name="ask" id="ask">
    <p>Recuerda terminar con un signo de interrogación (?)</p>
    <!--v-if-->
  </div>
</div>
<!--v-if-->
`;
